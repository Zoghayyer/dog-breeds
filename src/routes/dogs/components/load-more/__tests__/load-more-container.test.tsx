import React from 'react';
import { shallow } from 'enzyme';
import {
  DispatchProps,
  LoadMoreContainer,
} from '../load-more-container';

describe('(Container) LoadMoreContainer', () => {
  let component;
  const defaultProps: DispatchProps = {
    imagesLimitIncrease: () => ({}),
  };

  const shallowLoadMoreContainer = (extendedProps: Partial<DispatchProps> = {}) => {
    component = shallow(<LoadMoreContainer {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowLoadMoreContainer());

  it('renders LoadMoreView component', () => {
    expect(component).toMatchSnapshot();
  });

  describe('(Method) handleClick', () => {
    it('calls imagesLimitIncrease', () => {
      const imagesLimitIncreaseSpy = jest.fn();
      const props = {
        ...defaultProps,
        imagesLimitIncrease: imagesLimitIncreaseSpy,
      };
      component = new LoadMoreContainer(props);

      component.handleClick();

      expect(imagesLimitIncreaseSpy).toBeCalled();
    });
  });
});
