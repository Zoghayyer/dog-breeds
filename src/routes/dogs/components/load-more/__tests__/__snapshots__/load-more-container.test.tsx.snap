// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) LoadMoreContainer renders LoadMoreView component 1`] = `
<LoadMoreView
  handleClick={[Function]}
/>
`;
